src/test/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrderingIntegrationTest.java

```diff
@@ -0,0 +1,143 @@
+                {getTestFile(REVERSE_ORDERED_SAMPLE_MAP), getTestFile(BADLY_SORTED1000_BATCH_SIZE50_VCF), 10000}, //larger batch size than number of samples
+                {getTestFile(REVERSE_ORDERED_SAMPLE_MAP), getTestFile(BADLY_SORTED1000_BATCH_SIZE50_VCF), 0}, // batch size 0
+                {getTestFile("mismatched.sample_map"), getTestFile(BADLY_SORTED1000_BATCH_SIZE50_VCF), 50} //larger batch size than number of samples
+        };
+    }
+
+
+    /**
+     * this was used to generate the vcf files files used in {@link #testUnshufflingRestoresCorrectSamples(File, File, int)}
+     * it's kept around in case we need to generate new test cases or examine how they were made
+     */
+    @Test(enabled = false)
+    public void generateVCFs() throws IOException {
+        final File file = writeManyVCFs(1000);
+        System.out.println(file);
+    }
+
+    /**
+     *   test files were created by generating mangled files using the {@link #writeManyVCFs(int)} with parameter 1000
+     *   these were then imported to GenomicsDB using GATK 4.beta.5-66-g9609cb3-SNAPSHOT which is before the fix for GenomicsDBImport
+     *   sample name ordering was applied
+     *
+     *   the combined vcfs were then extracted using SelectVariants, these are the badlySorted1000 files
+     *   each vcf contains a single record with 1000 genotypes that have a single format field "SN: sample name"
+     *   these were created with SN's that matched their assigned sample names, but the badly sorted versions are shuffled
+     */
+    @DataProvider
+    public Object[][] getBadlySortedFiles(){
+        return new Object[][]{
+                {getTestFile(REVERSE_ORDERED_SAMPLE_MAP), getTestFile(BADLY_SORTED1000_BATCH_SIZE50_VCF), 50},
+                {getTestFile(REVERSE_ORDERED_SAMPLE_MAP), getTestFile("badlySorted1000-batch-size50.vcf.gz"), 50},
+                {getTestFile(REVERSE_ORDERED_SAMPLE_MAP), getTestFile("badlySorted1000-batch-size13.vcf"), 13},
+        };
+    }
+
+    @Test(dataProvider = "getBadlySortedFiles")
+    public void testUnshufflingRestoresCorrectSamples(final File sampleMap, final File shuffled, final int batchSize) throws IOException {
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        final File fixed = createTempFile("fixed_", ".vcf");
+        args.addVCF(shuffled)
+                .addFileArgument(GenomicsDBImport.SAMPLE_NAME_MAP_LONG_NAME, sampleMap)
+                .addArgument(GenomicsDBImport.BATCHSIZE_ARG_NAME, String.valueOf(batchSize))
+                .addBooleanArgument(FixCallSetSampleOrdering.SKIP_PROMPT_LONG_NAME, true)
+                .addOutput(fixed);
+        runCommandLine(args);
+
+        try(final FeatureReader<VariantContext> reader =
+                AbstractFeatureReader.getFeatureReader(fixed.getAbsolutePath(), new VCFCodec(), true)){
+            final VariantContext vc = reader.iterator().next();
+            vc.getGenotypes().iterator().forEachRemaining( g ->
```

Comment:
Add an assert before this check that you have the correct number of genotypes.

Reply:
good idea, added